File: java/src/com/db4o/dg2db4o/chapter7/Customer.java

package com.db4o.dg2db4o.chapter7;

public class Customer extends AbstractPerson{
    
    //static String _classVar = "CLASSVAR";
    Address _address;
    int _age;
    
    public Customer(String name, String phoneNumber, String email, Address address, int age) {
        super(name,phoneNumber,email);
        _address = address;
        _age = age;
        address.setCustomer(this);
    }
    
    public void sendMail(String fromAddress, String subject, String content) {
        String toAddress = super.getEmail();
        String sentContent = content + "COMPANY DISCLAIMER...";
        Emailer mailer = new Emailer();
        mailer.sendMail(fromAddress, toAddress, subject, content);
    }
    
    public Address getAddress() {
        return _address;
    }
    public void setAddress(Address value) {
        _address = value;
        value.setCustomer(this);
    }
    
    public String toString() {
        return super.getName() + " (Customer)";
    }

   
}

File: csharp/Chapter07/Customer.cs


namespace com.db4o.dg2db4o.chapter7
{
    class Customer : Person
    {
        Address _address;
        
        public Customer(string name, string phoneNumber, string email, Address address) : base(name,phoneNumber,email)
        {
            _address = address;
    //        address.Customer = this;
        }

        public override void SendMail(string fromAddress, string subject, string content)
        {
            string toAddress = base.Email;
            string sentContent = content + "COMPANY DISCLAIMER...";
            System.Net.Mail.SmtpClient mailer = new System.Net.Mail.SmtpClient();
            mailer.Send(fromAddress, toAddress, subject, content);
        }

        public Address Address
        {
            get
            {
                return _address;
            }
            set
            {
                _address = value;
                //value.Customer = this;
            }
        }

        public override string ToString()
        {
            return base.Name + " (Customer)";
        }
    }
}
